.github/workflows/build.yml

```diff
@@ -42,6 +42,83 @@ jobs:
             -Psonatype.username=${{ secrets.SONATYPE_USERNAME }}
             -Psonatype.password=${{ secrets.SONATYPE_PASSWORD }}
 
+  unit-tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+      - uses: gradle/wrapper-validation-action@v1
+      - name: Unit tests
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: testDebugUnitTest
+      - name: Upload failed unit test artifacts
+        if: failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: unit-test-failures
+          path: '**/build/reports'
+
+  instrumentation-tests:
+    name: Instrumentation tests
+    runs-on: macOS-latest
+    timeout-minutes: 20
+    strategy:
+      fail-fast: false
+      matrix:
```

Comment:
Do you want a matrix here? Only one API level is used.
AFAIK we do not use any Android OS dependent APIs. If we even use them, we do not have tests to cover a behaviour or just relying to AndroidX compatibility libraries.

Reply:
happy to remove it. thought it might be wise just in case we decided to go down that path.